lib/WebPackWrap.js

```diff
@@ -32,114 +27,131 @@ class WebPackGenerator {
-            ) {
-                // we should not use externalJS in modern style modules
-                resolve();
-                return;
-            }
-            let visualJsName = tsconfig.compilerOptions.out;
-            let visualJSFileContent = "";
-            let visualJSFilePath = visualPackage.buildPath(visualJsName);
-            const pbivizJsonPath = visualPackage.buildPath('pbiviz.json');
-            const pbiviz = require(pbivizJsonPath);
-            
-            visualJSFileContent += "\n" + fs.readFileSync(visualJSFilePath, { encoding: encoding });
-            visualJSFileContent =  "\nvar powerbi = globalPowerbi;\n" + visualJSFileContent;
-            visualJSFileContent += "\nmodule.exports = { powerbi };";
-            fs.writeFileSync(
-                visualJSFilePath,
-                visualJSFileContent
-            );
-            resolve();
-        });
+    static async appendExportPowerBINameSpace(visualPackage, tsconfig) { // jshint ignore:line
+        if (tsconfig.compilerOptions.outDir) {
+            // we should not use externalJS in modern style modules
+            return;
+        }
+        let visualJsName = tsconfig.compilerOptions.out;
+        let visualJSFileContent = "";
+        let visualJSFilePath = visualPackage.buildPath(visualJsName);
+        
+        visualJSFileContent += "\n" + fs.readFileSync(visualJSFilePath, { encoding: encoding });
+        visualJSFileContent =  "\nvar powerbi = globalPowerbi;\n" + visualJSFileContent;
+        visualJSFileContent += "\nmodule.exports = { powerbi };";
+        fs.writeFileSync(
+            visualJSFilePath,
+            visualJSFileContent
+        );
     }
 
-    static prepareWebPackConfig(visualPackage, options, tsconfig) {
-        return new Promise((resolve, reject) => {
-            const pbivizJsonPath = visualPackage.buildPath('pbiviz.json');
-            const pbiviz = require(pbivizJsonPath);
-            const capabliliesPath = pbiviz.capabilities;
-            const capabliliesFile = require(path.join(process.cwd(), capabliliesPath));
-            const webpackConfig = require('./webpack.config');
-            const visualJsName = tsconfig.compilerOptions.out;
-            const visualJsOutDir = tsconfig.compilerOptions.outDir;
-            const visualJSFilePath = visualPackage.buildPath(visualJsName || visualJsOutDir);
-            let externalJSFiles = [];
-            let externalJSFilesContent = "";
-            let externalJSFilesPath = path.join(visualPackage.basePath, config.build.precompileFolder, "externalJS.js");
-            if (pbiviz.externalJS) {
-                for (let file in pbiviz.externalJS) {
-                    externalJSFilesContent += "\n" + fs.readFileSync("./" +  pbiviz.externalJS[file], { encoding: encoding });
-                    externalJSFiles.push(path.join(visualPackage.basePath, pbiviz.externalJS[file]));
-                }
+    static async loadAPIPackage() { // jshint ignore:line
```

Comment:
You can use try catch syntax instead using fs because require throws an exception if module is not found